components/TextFields/tests/unit/ContainedInputView/MDCBaseTextFieldTests.m

```diff
@@ -99,4 +101,255 @@ - (void)testTrailingViewModeRTL {
+  [textField becomeFirstResponder];
+  [self forceLayoutOfView:textField];
+  XCTAssertNil(textField.leadingView);
+  XCTAssertNil(textField.trailingView);
+}
+
+- (void)testOnlyleadingViewWithViewModeUnlessEditing {
+  MDCBaseTextField *textField =
+      [self createTextFieldWithLeadingView:YES
+                              trailingView:NO
+                                  viewMode:UITextFieldViewModeUnlessEditing];
+  [self forceLayoutOfView:textField];
+  CGRect leadingFrameNotEditing = textField.leftView.frame;
+  XCTAssertFalse(textField.leadingView.hidden);
+  [textField becomeFirstResponder];
+  [self forceLayoutOfView:textField];
+  CGRect leadingFrameEditing = textField.leftView.frame;
+  XCTAssertTrue(textField.leadingView.hidden);
+  XCTAssertFalse([self CGRect:leadingFrameNotEditing isEqualToCGRect:leadingFrameEditing]);
+}
+
+- (void)testOnlytrailingViewWithViewModeUnlessEditing {
+  MDCBaseTextField *textField =
+      [self createTextFieldWithLeadingView:NO
+                              trailingView:YES
+                                  viewMode:UITextFieldViewModeUnlessEditing];
+  [self forceLayoutOfView:textField];
+  CGRect trailingFrameNotEditing = textField.rightView.frame;
+  XCTAssertFalse(textField.trailingView.hidden);
+  [textField becomeFirstResponder];
+  [self forceLayoutOfView:textField];
+  CGRect trailingFrameEditing = textField.rightView.frame;
+  XCTAssertTrue(textField.trailingView.hidden);
+  XCTAssertFalse([self CGRect:trailingFrameNotEditing isEqualToCGRect:trailingFrameEditing]);
+}
+
+- (void)testLeadingAndTrailingViewsWithViewModeUnlessEditing {
+  MDCBaseTextField *textField =
+      [self createTextFieldWithLeadingView:YES
+                              trailingView:YES
+                                  viewMode:UITextFieldViewModeUnlessEditing];
+  [self forceLayoutOfView:textField];
+  CGRect leadingFrameNotEditing = textField.leftView.frame;
+  CGRect trailingFrameNotEditing = textField.rightView.frame;
+  XCTAssertFalse(textField.leadingView.hidden);
+  XCTAssertFalse(textField.trailingView.hidden);
+  [textField becomeFirstResponder];
+  [self forceLayoutOfView:textField];
+  CGRect leadingFrameEditing = textField.leftView.frame;
+  CGRect trailingFrameEditing = textField.rightView.frame;
+  XCTAssertTrue(textField.leadingView.hidden);
+  XCTAssertTrue(textField.trailingView.hidden);
+  XCTAssertFalse([self CGRect:leadingFrameNotEditing isEqualToCGRect:leadingFrameEditing]);
+  XCTAssertFalse([self CGRect:trailingFrameNotEditing isEqualToCGRect:trailingFrameEditing]);
+}
+
+- (void)testNoLeadingOrTrailingViewWithViewModeUnlessEditing {
+  MDCBaseTextField *textField =
+      [self createTextFieldWithLeadingView:NO
+                              trailingView:NO
+                                  viewMode:UITextFieldViewModeUnlessEditing];
+  [self forceLayoutOfView:textField];
+  XCTAssertNil(textField.leadingView);
+  XCTAssertNil(textField.trailingView);
+}
```

Comment:
Please update the tests to follow our convention of "given/when/then" statements.

Reply:
Done.